src/main.cc

```diff
@@ -178,6 +178,39 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file) {
+		say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+		tarantool_exit(0);
+		return;
+	}
+
+	/**
+	 * Setting prompt explicitly every time, cause of
+	 * need to return from search mode.
+	 */
+	rl_set_prompt("tarantool> ");
```

Comment:
This looks fragile, because the prompt string is configured in [another file](https://github.com/tarantool/tarantool/blob/a33f3cc7f1bf6020de95d144d63050894d2d0716/src/box/lua/console.lua#L689) so changing the prompt or allowing the user to set it to an arbitrary string would result in a discrepancy.

Reply:
Fixed. For now there is no hardcoded prompt setting.